packages/dht/src/dht/PeerManager.ts

```diff
@@ -0,0 +1,339 @@
+/* eslint-disable @typescript-eslint/parameter-properties, class-methods-use-this */
+
+import KBucket from 'k-bucket'
+import { PeerDescriptor } from '../proto/packages/dht/protos/DhtRpc'
+import { DhtPeer } from './DhtPeer'
+import { IPeerManager } from './IPeerManager'
+import { SortedContactList } from './contact/SortedContactList'
+import { RandomContactList } from './contact/RandomContactList'
+import { PeerID, PeerIDKey } from '../helpers/PeerID'
+import { Logger } from '@streamr/utils'
+import { keyFromPeerDescriptor, peerIdFromPeerDescriptor } from '../helpers/peerIdFromPeerDescriptor'
+import { ConnectionManager } from '../connection/ConnectionManager'
+import EventEmitter from 'eventemitter3'
+import { DisconnectionType } from '../transport/ITransport'
+
+const logger = new Logger(module)
+
+export interface PeerManagerConfig {
+    numberOfNodesPerKBucket: number
+    maxNeighborListSize: number
+    getClosestContactsLimit: number
+    ownPeerId: PeerID
+    connectionManager: ConnectionManager
```

Comment:
The `connectionManager` is used to unlock connections, but the locking is done elsewhere? Should we move the unlocking also to that same place? (The connection management seems to a separate concern, not part of the functionality of this class.)

Reply:
I would agree with putting the locking functionality in same place. @juslesan opinions about this?

Reply:
Seems that the PeerManager is only calling `weakLockConnection` and `weakUnlockConnection` from the ConnectionManager. The PeerManager could only simply take the `ConnectionLocker`-interface as a parameter since other ConnectionManager logic seems to not be relevant here.

Reply:
Good idea, I'll switch to passing the interface